app/models/service_provider_request.rb

```diff
@@ -0,0 +1,5 @@
+class ServiceProviderRequest < ActiveRecord::Base
+  def self.find_by(*args)
```

Comment:
I think overriding `find_by` can lead to confusing behavior especially since will never return `nil`.

What about making a new method?

```ruby
def self.find_by_or_null(*args)
  find_by(*args) || NullServiceProviderRequest.new
end
```